src/baal/bayesian/consistent_dropout.py

```diff
@@ -0,0 +1,143 @@
+import copy
+
+import torch
+from torch.nn import functional as F
+from torch.nn.modules.dropout import _DropoutNd
+
+
+class ConsistentDropout(_DropoutNd):
+    """
+    ConsistentDropout is useful when doing research.
+    It guarantees that while the masks are the same between batches
+    during inference. The masks are different inside the batch.
+
+    This is slower than using regular Dropout, but it is useful
+    when you want to use the same set of weights for each sample used in inference.
+    
+    From BatchBALD (Kirsch et al, 2019), this is necessary to use BatchBALD and remove noise 
+    from the prediction.
+
+    Args:
+        p (float): probability of an element to be zeroed. Default: 0.5
+	
+	Notes:
+		For optimal results, you should use a batch size of one
+		during inference time.
+		Furthermore, to guarantee that each sample uses the same
+		set of weights,
+		you must use `replicate_in_memory=True` in ModelWrapper
+		, which is the default.  
+    """
+
+    def __init__(self, p=0.5):
+        super().__init__(p=p, inplace=False)
+        self.reset()
+
+    def forward(self, x):
+        if self.training:
+            return F.dropout(x, self.p, training=True, inplace=False)
+        else:
+            if self._mask is None or self._mask.shape != x.shape:
+                self._mask = self._make_mask(x)
+            return torch.mul(x, self._mask)
+
+    def _make_mask(self, x):
+        return F.dropout(torch.ones_like(x, device=x.device), self.p, training=True)
```

Comment:
Could you set the batch dimension to 1 here, so it will broadcast?  (Or you can also drop it which should have the same effect IIRC.)

We want the masks to be consistent within the batch, too.

Reply:
(Just our discussion on Slack to have everything at the same place)
In the note, we said that we expected a batch size of 1 and to use replicate_in_mem=True.

I am trying to come up with something to remove this constraint or better show it. I'm open to suggestion :)